### ISSUE TITLE
Server side copy starts failing with smbj 0.11.1

### ISSUE THREAD

**Issue by jansohn** [hierynomus/smbj#654]
Our working code for server side copy on smbj 0.10.0 starts failing with the following error when switching to the latest version 0.11.1:
```
com.hierynomus.mssmb2.SMBApiException: STATUS_NOT_SUPPORTED (0xc00000bb): IOCTL failed for SMB2FileId{persistentHandle=e3 09 ae 7a a6 05 2c 3b}
	at com.hierynomus.smbj.share.Share.receive(Share.java:397)
	at com.hierynomus.smbj.share.Share.ioctl(Share.java:332)
	at com.hierynomus.smbj.share.Share.ioctl(Share.java:315)
	at com.hierynomus.smbj.share.DiskEntry.ioctl(DiskEntry.java:184)
	at com.hierynomus.smbj.share.File.getResumeKey(File.java:277)
	at com.hierynomus.smbj.share.File.remoteFileCopy(File.java:223)
	at com.hierynomus.smbj.share.File.remoteCopyTo(File.java:216)
	at com.hierynomus.smbj.share.File.remoteCopyTo(File.java:202)
```

This happens with `SMB2Dialect.SMB_2_1` on a NetApp DFS share. I would blame NetApp normally but since it is working with 0.10.0 this seems to be a regression in smbj. I cannot reproduce the problem on a Windows Server 2016 share btw...

**Comment by jansohn** [comment]
The difference is that in `File.remoteCopyTo(File destination)` (https://github.com/hierynomus/smbj/blob/master/src/main/java/com/hierynomus/smbj/share/File.java#L197) the check returns `true` for 0.10.0 (not the same disk share) and `false` for 0.11.1 (same disk share).

We have the same check in our code before actually calling `remoteCopyTo()` which is why I did not notice that with 0.10.0 we are actually not performing the server side copy at all but the fallback "manual" copy instead.

I guess that the behavior we are seeing in 0.10.0 is correct because with 0.11.1 get the clear response from the server that the server side copy is not supported.

So all that remains is to find the reason why with 0.11.1 the disk share does not change anymore when opening the files.

Sample code for easier reproduction:
```
Configuration config = new Configuration();
config.readPropertiesFromFile();

AuthenticationContext authContext = new AuthenticationContext(config.getSambaConfig().getUsername(),
        config.getSambaConfig().getPassword().toCharArray(),
        config.getSambaConfig().getDomain());

SmbConfig clientConfig = SmbConfig.builder()
        .withDfsEnabled(true)
        .withSecurityProvider(new BCSecurityProvider())
        .withDialects(SMB2Dialect.SMB_2_1)
        .withMultiProtocolNegotiate(true)
        .withSigningRequired(true)
        .build();

SmbPath remoteDir = SmbPath.parse("\\\\server03088.emea.my-company.com\\frd\\public\\transfer\\dfsMoveTest");
SmbPath remoteTargetDir = SmbPath.parse("\\\\server03088.emea.my-company.com\\frd\\public\\transfer\\dfsMovedTest");

SmbPath sourcePath = new SmbPath(remoteDir, "test.txt");
SmbPath targetPath = new SmbPath(remoteTargetDir, "test1.txt");

// Connect to share
try (SMBClient client = new SMBClient(clientConfig);
        Connection connection = client.connect(remoteDir.getHostname());
        Session session = connection.authenticate(authContext);
        DiskShare diskShare = (DiskShare) session.connectShare(remoteDir.getShareName()))
{
    log.info("Connected to {} with Samba protocol {}", remoteDir.getHostname(), connection.getNegotiatedProtocol().getDialect());

    // move the file
    try (File fileToCopy = diskShare.openFile(sourcePath.getPath(),
            EnumSet.of(AccessMask.GENERIC_READ),
            null,
            SMB2ShareAccess.ALL,
            SMB2CreateDisposition.FILE_OPEN,
            null);
            File fileToWrite = diskShare.openFile(targetPath.getPath(),
                    EnumSet.of(AccessMask.GENERIC_WRITE),
                    EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
                    EnumSet.of(SMB2ShareAccess.FILE_SHARE_WRITE),
                    SMB2CreateDisposition.FILE_OVERWRITE_IF,
                    EnumSet.noneOf(SMB2CreateOptions.class)))
    {
        log.info("{} == {} => {}", fileToCopy.getDiskShare(), fileToWrite.getDiskShare(), fileToCopy.getDiskShare().equals(fileToWrite
                .getDiskShare()));
        fileToCopy.remoteCopyTo(fileToWrite);
    }
}
```

**Comment by vrr6** [comment]
Hi @hierynomus ,

We are also facing the exact issue please check .

temporarily I changed my application code  to use SmbFiles.copy method.

thanks,
R Ramarajan.

**Comment by mnadeem** [comment]
@hierynomus 

I am also getting similar issue, with **0.11.3**


```
com.hierynomus.mssmb2.SMBApiException: STATUS_NOT_SUPPORTED (0xc00000bb): IOCTL failed for SMB2FileId{persistentHandle=01 e0 04 8e 5c 78 f2 79}
```

**Comment by hierynomus** [comment]
I think this is fixed with the merge of #683

**Comment by zhanghai** [comment]
Thanks for fixing this - do you have plan to create a new release with this fix recently?

**Comment by schmolkman** [comment]
Hey @hierynomus, we are currently blocked by this issue as well, since performing a remote copying of the file is an essential part of our workflow. Using a "download -> then upload" kind of a workaround is really problematic for performance reasons (large files).

Is there a plan/timeline regarding a new release the will include this fix?

Probably there are more people who can't wait to get their hands on a brand new v,0.11.4 😉 

Thank you in advance 😃

**Comment by hierynomus** [comment]
v0.11.5 was released! (Something went wrong with 0.11.4, so skipped a version ;))

**Comment by schmolkman** [comment]
**_Thanks a lot_** @hierynomus  for the quick response and the new release with the fix ❗  👍 
We are looking forward to taking the new v0.11,5 for a spin in our next sprint 😉
